<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClientWindow</class>
 <widget class="QMainWindow" name="ClientWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QWidget{background:#ffffff;color:#1f1f1f;font:10pt &quot;Microsoft YaHei&quot;;}
QLineEdit{background:#ffffff;color:#1f1f1f;border:1px solid #cfcfcf;border-radius:6px;padding:6px 8px;}
QPlainTextEdit{background:#ffffff;color:#1f1f1f;border:1px solid #cfcfcf;border-radius:8px;padding:6px;}
QListWidget{background:#ffffff;color:#1f1f1f;border:1px solid #cfcfcf;border-radius:8px;}
QPushButton{background:#00a67d;color:#ffffff;border:none;border-radius:8px;padding:10px 14px;font-weight:600;}
QPushButton:hover{background:#009a74;}
QPushButton:pressed{background:#008a67;}
QPushButton#pushButtonExit{background:#e74c3c;}
QPushButton#pushButtonExit:hover{background:#d64535;}
QPushButton#pushButtonExit:pressed{background:#c0392b;}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayoutRoot">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QLabel" name="labelTitle">
      <property name="minimumHeight">
       <number>60</number>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel#labelTitle{background:#00a67d;color:white;font: 20pt &quot;Microsoft YaHei&quot;;padding-left:12px;}</string>
      </property>
      <property name="text">
       <string>多用户聊天室（实验4）</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="pageLogin">
       <layout class="QVBoxLayout" name="verticalLayoutLoginPage">
        <item>
         <spacer name="verticalSpacerTop">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>140</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutLoginCenter">
          <item>
           <spacer name="horizontalSpacerLeft">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>200</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QFrame" name="frameLogin">
            <property name="minimumSize">
             <size>
              <width>360</width>
              <height>200</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">QFrame#frameLogin{border:1px solid #bdbdbd;border-radius:12px;background:#ffffff;}</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayoutLoginCard">
             <property name="leftMargin">
              <number>24</number>
             </property>
             <property name="topMargin">
              <number>24</number>
             </property>
             <property name="rightMargin">
              <number>24</number>
             </property>
             <property name="bottomMargin">
              <number>24</number>
             </property>
             <item>
              <layout class="QFormLayout" name="formLayoutLogin">
               <item row="0" column="0">
                <widget class="QLabel" name="labelServerIp">
                 <property name="text">
                  <string>服务器IP地址：</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="lineEditHost">
                 <property name="text">
                  <string>127.0.0.1</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="labelNick">
                 <property name="text">
                  <string>昵称：</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="lineEditName">
                 <property name="text">
                  <string>user</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonLogin">
               <property name="text">
                <string>登录聊天室</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacerRight">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>200</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacerBottom">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>200</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="pageChat">
       <layout class="QVBoxLayout" name="verticalLayoutChatPage">
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitterChat">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QPlainTextEdit" name="plainTextEditChat">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QListWidget" name="listWidgetUsers">
           <property name="minimumWidth">
            <number>140</number>
           </property>
           <property name="maximumWidth">
            <number>260</number>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutSend">
          <item>
           <widget class="QLineEdit" name="lineEditMessage">
            <property name="placeholderText">
             <string>输入消息...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonSend">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonExit">
            <property name="text">
             <string>退出</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
